### PR TITLE
OneLogin event collector

### PR DESCRIPTION
**Pull request by yaakovpraisler**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [link to the issue](https://jira-hq.paloaltonetworks.local/browse/CIAC-3242)

## Description
OneLogin event collector.

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

File: Packs/OneLogin/Integrations/OneLoginEventCollector/OneLoginEventCollector.py [hunk 1]
```diff
@@ -0,0 +1,367 @@
...
+            expired_token_date_time = parse(int_context['expired_token_time'], settings={'TIMEZONE': 'UTC'})
+            if expired_token_date_time and datetime.now(timezone.utc) - expired_token_date_time < timedelta(minutes=590):
+                demisto.debug("Found access token in integration context.")
+                return int_context.get('access_token')
+
+        demisto.debug("Access token has expired, sending request to get a new access token.")
+        access_token = self.get_access_token_request()
+        set_integration_context({
+            'access_token': access_token,
+            'expired_token_time': datetime.utcnow().strftime(DATE_FORMAT)
```
Comment (dantavori) [inline Packs/OneLogin/Integrations/OneLoginEventCollector/OneLoginEventCollector.py:166]: this line makes it look like expiry time is now (confusing), I'd do:
```suggestion
            'expired_token_time': datetime.utcnow().strftime(DATE_FORMAT) + timedelta(minutes=590)
```
and remove the timedelta above

File: Packs/OneLogin/Integrations/OneLoginEventCollector/OneLoginEventCollector.py [hunk 2]
```diff
@@ -0,0 +1,367 @@
...
+                break
+
+        except DemistoException as e:
+            if not e.res or e.res.status_code != 429:
+                raise e
+            demisto.debug('Reached API rate limit, storing last used cursor.')
+            cursor = query_params['after_cursor']
+
+        since = query_params['since'] if cursor or not aggregated_events else increase_time_with_second(
+            aggregated_events[-1].get('created_at'))
```
Comment (dantavori) [inline Packs/OneLogin/Integrations/OneLoginEventCollector/OneLoginEventCollector.py:242]: is it possible to miss events like this?

File: Packs/OneLogin/Integrations/OneLoginEventCollector/OneLoginEventCollector.py [hunk 3]
```diff
@@ -0,0 +1,367 @@
...
+
+
+def get_event_type_name(event_type_id, event_types):
+    """
+    Returns the event type name by the given event type ID.
+    """
+    if event_type_id in event_types:
+        return event_types[event_type_id]
+
+    return None
```
Comment (dantavori) [inline Packs/OneLogin/Integrations/OneLoginEventCollector/OneLoginEventCollector.py:55]: will do the same
```suggestion
    return event_types.get(event_type_id)
```
don't really need a method for this though

**Review by dantavori — COMMENTED** [review]
looks great! some minor comments.
please assign a doc reviewer, I will approve the PR later (waiting for a green light from @evisochek)

**Comment by ShirleyDenkberg** [comment]
@yaakovpraisler Need to add content for the Pack Readme.
@dantavori Doc review completed.

File: Packs/OneLogin/Integrations/OneLoginEventCollector/OneLoginEventCollector.py (hunk 1 above)
Comment (yaakovpraisler) [reply Packs/OneLogin/Integrations/OneLoginEventCollector/OneLoginEventCollector.py]: done

File: Packs/OneLogin/Integrations/OneLoginEventCollector/OneLoginEventCollector.py (hunk 2 above)
Comment (dantavori) [reply Packs/OneLogin/Integrations/OneLoginEventCollector/OneLoginEventCollector.py:231]: is it possible to have multiple events at the exact same time in the product? then if we fetch only some of them in the first run, we won't fetch the rest of them in the following run

File: Packs/OneLogin/ModelingRules/OneLoginModelingRules/OneLoginModelingRules.yml [hunk 4]
```diff
@@ -0,0 +1,7 @@
+name: OneLogin Modeling Rule
+id: onelogin_modeling_rule
+fromversion: 6.8.0
+toversion: 6.10.0
```
Comment (evisochek) [inline Packs/OneLogin/ModelingRules/OneLoginModelingRules/OneLoginModelingRules.yml:4]: @yaakovpraisler should be to 6.9.9

Comment (yaakovpraisler) [reply Packs/OneLogin/ModelingRules/OneLoginModelingRules/OneLoginModelingRules.yml]: fixed

**Review by dantavori — APPROVED** [review]

**Pull request merged**